### DISCUSSION TITLE
switchChain function received me an error: TypeError: Cannot destructure property 'addEthereumChainParameter' of 'parameters' as it is undefined.

### DISCUSSION THREAD

**Discussion by RedirexStudio** [wevm/wagmi#4165] [Q&A]
I have an issue with switching my chain by click on button.
What i had do:
`import { switchChain } from "@wagmi/core";`

Then:
```
  const switchNetworkHandler = async () => {
    try {
      await switchChain({ chainId: bsc.id });
      console.log('Successfully switched to BSC');
    } catch (error) {
      console.error('Failed to switch to BSC:', error);
    }
  };
```

This handler i put into onClick={switchNetworkHandler} of my button.
But when i click on button i see this error:
![Screenshot_10](https://github.com/user-attachments/assets/21f33c5e-a6ed-4469-8be3-ef2d94acd2e1)

`TypeError: Cannot destructure property 'addEthereumChainParameter' of 'parameters' as it is undefined.`

Where i made mistake, help me please.
By the way: other methods, like: window.ethereum.request working fine. But i want do it through wagmi...

Dependencies:
```
"wagmi": "^2.12.0",
"viem": "^2.17.10",
"@web3modal/wagmi": "^5.0.7"
```

**Comment by tmm** [comment]
You are missing the `config` parameter for [`switchChain`](https://wagmi.sh/core/api/actions/switchChain)

```diff
- await switchChain({ chainId: bsc.id });
+ await switchChain(config, { chainId: bsc.id });
```

**Reply by RedirexStudio** [reply]
I forgot to added it to my message. I have config and i already added it to my switchChain. But it doesn't give any result.
This is my config:
```
  const config = createConfig({
    chains: [bsc],
    transports: {
      [bsc.id]: http(),
    },
  })
```

So, finaly i have this code:
 ```
 const switchNetworkHandler = async () => {
    try {
      await switchChain(config, { chainId: bsc.id });
      console.log('Successfully switched to BSC');
    } catch (error) {
      console.error('Failed to switch to BSC:', error);
    }
  };
```

I can share you my full code if it's will help.

.. but received the same error :(

**Comment by RedirexStudio** [comment]
@tmm it's not resolved, don't close this topic please

**Reply by tmm** [reply]
If you think there is a bug, open an issue with a minimal reproduction.
